server/src/signin-button.js

```diff
@@ -11,25 +11,29 @@ exports.SignInButton = class SignInButton extends React.Component {
   render() {
     if (this.props.isAuthenticated) {
       return <Localized id="settingsButton">
-        <a className="settings-button" href="/settings" aria-label="Settings">
+        <a className="settings-button" onClick={ this.clickHandler.bind(this) } aria-label="Settings">
```

Comment:
The markup is still incorrect with this change.  Try changing the `if` condition above to base on a state property that derives from `this.props.isAuthenticated`.

Reply:
Updated to use state though it doesn't help with markup. Is there any way to force re-render